core/new-gui/src/app/workspace/component/workflow-editor/workflow-editor.component.ts

```diff
@@ -183,6 +185,22 @@ export class WorkflowEditorComponent implements AfterViewInit {
         }
       );
   }
+  /**
+   * if the operator is valid , the border of the box will be green
+   */
+  private handleOperatorValidation(): void {
+
+    this.validationWorkflowService.getOperatorValidationStream()
+      .subscribe( value => {
+        if (value.status) {
+          // find the operator element by its ID and change the css style of the stroke
+          this.getJointPaper().getModelById(value.operatorID).attr('rect/stroke', 'green');
+        } else {
+          this.getJointPaper().getModelById(value.operatorID).attr('rect/stroke', 'red');
+        }
```

Comment:
rather than changing the style here directly, write the detail of the change in `joint-ui.service` to isolate the detail of style changes on workflow paper to only one service. Doing this make our code easier to manage as each service is only in-charge of doing one task, and component will not need to know any details regarding that task (just call the method defined in the service).